runner/daemon/src/mill/daemon/Watching.scala

```diff
@@ -109,13 +250,23 @@ object Watching {
     statWatchWait0()
   }
 
-  def validate(w: Watchable) = poll(w) == signature(w)
-  def poll(w: Watchable) = w match {
+  /** @return true if the watchable did not change. */
+  inline def validate(w: Watchable.Pollable): Boolean = validateAnyWatchable(w)
+
+  /**
+   * As [[validate]] but accepts any [[Watchable]] for the cases when we do not want to use a notification system.
+   *
+   * Normally you should use [[validate]] so that types would guide your implementation.
+   */
+  def validateAnyWatchable(w: Watchable): Boolean = poll(w) == signature(w)
```

Comment:
This name is pretty weird; it wasn't great before but if we're going to change it we should use a better one. Maybe `def checkForUpdate` or something?